src/components/MaterialUI/Inputs/Switch.test.js

```diff
@@ -60,13 +79,40 @@ describe("Switch Component", () => {
 		expect(update, "to have calls satisfying", [{ args: [true] }]);
 	});
 
+	it("Checkbox component not handles check if it's read only", async () => {
+		const switchProps = new SwitchProps();
+
+		switchProps.set(SwitchProps.propNames.update, update);
+		switchProps.set(SwitchProps.propNames.value, false);
+		switchProps.set(SwitchProps.propNames.readOnly, true);
+
+		ReactDOM.render(
+			<IntlProvider locale="en-US">
+				<Switch switchProps={switchProps} />
+			</IntlProvider>,
+			container,
+		);
+
+		const clickEvent = document.createEvent("MouseEvents");
+		clickEvent.initEvent("click", true, false);
+
+		const element = container.querySelector(".MuiSwitch-input ");
+		element.dispatchEvent(clickEvent);
+		expect(update, "to have calls satisfying", []);
```

Comment:
don't use that approach for render tests. You enzyme to render and pick up necessary dom elements and invoke events on them